kasa/smartprotocol.py

```diff
@@ -157,15 +147,50 @@ async def _execute_query(self, request: Union[str, Dict], retry_count: int) -> D
             self._host,
             _LOGGER.isEnabledFor(logging.DEBUG) and pf(smart_request),
         )
-        response_data = await self._transport.send(smart_request)
+        resp_dict = await self._transport.send(smart_request)
 
         _LOGGER.debug(
             "%s << %s",
             self._host,
-            _LOGGER.isEnabledFor(logging.DEBUG) and pf(response_data),
+            _LOGGER.isEnabledFor(logging.DEBUG) and pf(resp_dict),
         )
 
-        return response_data
+        self._handle_response_error_code(resp_dict)
+
+        if result := resp_dict.get("result"):
+            if is_multi:
```

Comment:
Invert the check to avoid one level of nesting, something like:
```
if not is_multi:
    return {smart_method: result}

<multiresponse handling>
```
In fact, the negative case for `result` availability could be done to de-nest it one step further:
```
if "result" not in resp_dict:  # should this even happen?!
    return {smart_method: None}
    
if not is_multi:
    # handle single commands
else:
    # handle multi
```
What do you think?

Reply:
I've dropped the `isMulti` as per your suggestion below and also reduced the nesting.